service/controller/resource/clusterrolebinding/update.go

```diff
@@ -21,7 +22,19 @@ func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange inte
 		// Create the cluster role bindings in the Kubernetes API.
 		for _, clusterRoleBinding := range clusterRoleBindingsToUpdate {
 			_, err := r.k8sClient.RbacV1().ClusterRoleBindings().Update(clusterRoleBinding)
-			if err != nil {
+			if isExternalFieldImmutableError(err) {
+				// We can't change a RoleRef, so create a new CRB which references the new RoleRef
+				newCRB := apiv1.ClusterRoleBinding{}
+
+				clusterRoleBinding.DeepCopyInto(&newCRB)
+				newCRB.SetName(clusterRoleBinding.Name + "-upgrading")
```

Comment:
We should change this back eventually somehow. Possibly automatically?

Reply:
Yeah agreed, I'm not sure yet exactly when it is "safe" to delete the old one. I can wait until all old pods are gone, but I don't know how/where to track this at the moment

Reply:
Actually, I don't think I can rename the existing CRB. So the process would be 1 create the new one with '-upgrading', 2 delete the old one, 3 create a new new one without the suffix, 4 delete the '-upgrading' CRB. I think that would result in a second roll of the nodes

Reply:
Otherwise I YOLO it and delete the old one before creating the new one

Reply:
After thinking about this a bit, the important thing is that we can't change the `ClusterRole` of a `ClusterRoleBinding` from `kvm-operator` to `kvm-operator-<version>` or from `kvm-operator-<previous version>` to `kvm-operator-<version>`. In short, this will happen any time the operator version is changed.

Instead of appending `-upgrading` to the name of the CRB and hitting this same issue on the next upgrade, what do you think of instead appending the `ClusterRole` name to it or a short hash thereof (maybe using crc32)? For TC with ID `abc12`, the CRB would look like `abc12-kvm-operator-<version>` or `abc12-<hash>`. Once the rollout of the new nodes is complete, we can then look for and delete any CRB matching `abc12*` other than the CRB currently in use. This is similar to how Kubernetes manages a `StatefulSet` based on names. We could alternatively select CRBs by label `giantswarm.io/cluster-id=abc12` to clean them up.

Reply:
I agree that creating one from e.g. the operator version would make a lot of sense.

Reply:
Yes also agreed. However it may not even be necessary. I just tested an update using a version which creates a temporary CRB, deletes the original, then creates the new CRB using the old name. So the end effect looks like it was updated in-place. The upgrade was successful and all TC pods are running, so initial impression is ok. Will update here momentarily, then we can decide if there is still a need to add versioned CRBs